Update mongo tests

king-alexander opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->
These changes update tests that connect to a mock Mongo database.

## 💭 Motivation and context ##

<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->
- Closes #50 

## 🧪 Testing ##

<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

I modified the tests that connect to a mock Mongo database to use the new connection method. After my changes, all tests passed locally.

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- These boxes should remain unchecked until the pull request has been -->
<!-- approved. -->

- [x] Revert dependencies to default branches.
- [x] Finalize version.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [ ] Create a release.

king-alexander committed:
Modify connection URIs

The 'mongomock://' URI is deprecated, so the connection URIs now start
with 'mongodb://'.

king-alexander committed:
Specify mongomock in connection

king-alexander committed:
Update valid configuration test

This test broke once the 'mongomock://' URI was no longer supported. I
had to rewrite the test to use the new connection method.

king-alexander committed:
Drop support for Python 3.6

The mongomock connection method introduced in MongoEngine 0.27.0 is not
recognized in Python 3.6.

king-alexander commented:
Should we drop support for Python 3.6?

king-alexander committed:
Require MongoEngine v0.27.0 or higher

jasonodoom reviewed (approved):
LGTM!

jsf9k commented on .github/workflows/build.yml:
```diff
@@ -111,14 +111,13 @@ jobs:
         include:
           - os: ubuntu-20.04
-            python-version: "3.6"
+            python-version: "3.7"
```
Why does 3.7 need to be run under Ubuntu 20.04?

king-alexander replied:
Pardon -- I wasn't familiar enough with workflows syntax when I made the change and assumed that was required. But I suppose only 3.6 had to be run under Ubuntu 20.04 to function properly. I edited [b586c69](https://github.com/cisagov/admiral/pull/47/commits/b586c69e8dda6e02eeef8b0d3f4f506e73ae483b) to remove this extra configuration.

jsf9k commented on .github/workflows/build.yml:
```diff
@@ -207,14 +206,13 @@ jobs:
         include:
           - os: ubuntu-20.04
-            python-version: "3.6"
+            python-version: "3.7"
```
Why does 3.7 need to be run under Ubuntu 20.04?

king-alexander replied:
See https://github.com/cisagov/admiral/pull/47#discussion_r1254703891.

jsf9k commented on .github/workflows/build.yml:
```diff
@@ -260,14 +258,13 @@ jobs:
         include:
           - os: ubuntu-20.04
-            python-version: "3.6"
+            python-version: "3.7"
```
Why does 3.7 need to be run under Ubuntu 20.04?

king-alexander replied:
See https://github.com/cisagov/admiral/pull/47#discussion_r1254703891.

jsf9k commented on tests/test_util.py:
```diff
@@ -32,5 +34,8 @@ def test_load_config_valid():
     """Test that a valid configuration connects to appropriate connections."""
-    # TODO: Update this connection method. See #50 for more details.
-    util.connect_from_config(valid_config)
+    connections = valid_config["connections"]
+    for alias in connections.keys():
+        connect(
+            host=connections[alias]["uri"], mongo_client_class=MongoClient, alias=alias
+        )
```
Would it be better to add a `mongo_client_class` argument to `util.connect_from_config()` with the appropriate default value so that it continues to function as expected if `mongo_client_class` is unspecified?  That way you could call `util.connect_from_config()` here and (1) slightly improve the test coverage and (2) DRY the code out a little by avoiding recreating the body of `util.connect_from_config()` in this test function.

king-alexander replied:
Great suggestion! I applied it in [9973720](https://github.com/cisagov/admiral/pull/47/commits/99737202bfa252c9077526b38daeb54fb3e07ac2) and [193ac57](https://github.com/cisagov/admiral/pull/47/commits/193ac57bcaff55fae669a045c04bbf47b30b5590).

jsf9k reviewed (commented):
I have a few questions/comments.

king-alexander committed:
Make the client class a connection argument

This commit augments `connect_from_config()` with a `client` argument to
determine which client class to use when called. The default client
class for MongoEngine (`pymongo.MongoClient`) is explicitly made the
default value so alternative client classes can be specified where
necessary.

king-alexander committed:
Use mock mongo client

Connect MongoEngine with mongomock to test connections from a
configuration.

jsf9k reviewed (commented):
These changes look good to me, but I'll wait for the GH Actions checks to pass before actually approving.

dav3r commented:
> These changes look good to me, but I'll wait for the GH Actions checks to pass before actually approving.

Yes, please let us know when the GH Actions all pass and then we will finalize our reviews.

king-alexander commented:
I will add a test to boost coverage and make the Coveralls check happy.

king-alexander committed:
Upgrade PyYAML

The 6.0 release contains a temporary workaround to a regression in
Cython 3.10.0a10 that throws an attribute error. See more details
upstream at https://github.com/yaml/pyyaml/issues/601 and
https://github.com/yaml/pyyaml/pull/702.

king-alexander committed:
Remove Docker Compose dependency

The Docker Compose package is causing a dependency conflict. And with
respect to #40, the package no longer belongs in this project.

king-alexander committed:
Add test subject

This fabricated subject covers domains that end in "fed.us" to boost
test coverage.

jsf9k reviewed (approved):
Strong work!

dav3r reviewed (approved):
Looks good! 👍 👍

king-alexander committed:
Bump version from 1.3.3 to 1.4.0
